StereoKitC/systems/render.cpp

```diff
@@ -158,12 +172,54 @@ void render_set_fov(float field_of_view_degrees) {
 
 ///////////////////////////////////////////
 
+void render_set_ortho_height(float render_ortho_viewport_height) {
+	render_ortho_viewport_height = render_ortho_viewport_height;
+	render_update_projection();
+}
+
+///////////////////////////////////////////
+
+void render_set_ortho_clip(float near_plane, float far_plane) {
+	render_ortho_near_clip = near_plane;
+	render_ortho_far_clip = far_plane;
+	render_update_projection();
+}
+
+///////////////////////////////////////////
+
+void render_set_projection(projection_ proj) {
+	render_projection_type = proj;
+	render_update_projection();
+}
+
+///////////////////////////////////////////
+
+projection_ render_get_projection_type() {
+	return render_projection_type;
+}
+
+///////////////////////////////////////////
+
+float render_get_ortho_view_height() {
+	return render_ortho_viewport_height;
+}
+
+///////////////////////////////////////////
+
 void render_update_projection() {
-	math_fast_to_matrix( XMMatrixPerspectiveFovRH(
-		render_fov * deg2rad, 
-		(float)sk_system_info().display_width/sk_system_info().display_height, 
-		render_clip_planes.x, 
-		render_clip_planes.y), &render_default_camera_proj);
+	if (render_projection_type == projection_perspective) {
+		math_fast_to_matrix(
+		    XMMatrixPerspectiveFovRH(render_fov * deg2rad,
+		                             (float)sk_system_info().display_width / sk_system_info().display_height,
+		                             render_clip_planes.x,
+		                             render_clip_planes.y),
+		    &render_default_camera_proj);
+	} else {
+		math_fast_to_matrix(
+		    XMMatrixOrthographicRH(
+		        (float)sk_system_info().display_width / sk_system_info().display_height, 1.0, 0, 2.0),
```

Comment:
Missed the clipping planes!

Reply:
indeed - also missed the viewport height 🙀